lntest/itest/lnd_channel-funding_fund_max_test.go

```diff
@@ -0,0 +1,296 @@
+		)
+
+		// Stop at the first failure. Mimic behavior of original test
+		// framework.
+		if !success {
+			break
+		}
+	}
+}
+
+// runTestCase runs a single test case asserting that test conditions are met.
+func runTestCase(net *lntest.NetworkHarness, ht *harnessTest,
+	t *testing.T, carol, dave *lntest.HarnessNode,
+	testCase *chanFundMaxTestCase) {
+
+	net.SendCoins(t, testCase.amount, carol)
+	defer func() {
+		if testCase.amount <= 2_000 {
+			// Add additional funds to sweep "dust" UTXO.
+			net.SendCoins(ht.t, 100_000, carol)
+		}
+
+		// Remove all funds from Carol.
+		sweepNodeWalletAndAssert(net, ht, carol)
+	}()
+
+	// The parameters to try opening the channel with.
+	chanParams := lntest.OpenChannelParams{
+		Amt:         0,
+		PushAmt:     testCase.pushAmt,
+		SatPerVByte: testCase.feeRate,
+		FundMax:     true,
+	}
+
+	// If we don't expect the channel opening to be
+	// successful, simply check for an error.
+	if testCase.chanOpenShouldFail {
+
+		_, err := net.OpenChannel(
+			carol, dave, chanParams,
+		)
+
+		if err == nil {
+			t.Errorf("should not open channel "+
+				"for: %v", testCase.name)
+		}
+
+		return
+	}
+
+	// Otherwise, if we expect to open a channel use the helper function.
+	chanPoint := openChannelAndAssert(
+		ht, net, carol, dave, chanParams,
+	)
+	defer func() {
+		// Close the channel between Carol and Dave, asserting
+		// that the channel has been properly closed on-chain.
+		closeChannelAndAssert(
+			ht, net, carol, chanPoint, false,
+		)
+	}()
+
+	cType, err := channelCommitType(carol, chanPoint)
+	if err != nil {
+		ht.Fatalf("unable to get channel type: %v", err)
+	}
+
+	// Carol's balance should be her amount subtracted by the
+	// commitment transaciton fee.
+	err = checkChannelBalance(
+		ht, carol, testCase.carolBalance-calcStaticFee(cType, 0),
+	)
+	if err != nil {
+		t.Errorf("carol's balance is wrong: %v", err)
+	}
+
+	// Ensure Dave's balance within the channel is equal to the push amount.
+	err = checkChannelBalance(ht, dave, testCase.pushAmt)
+	if err != nil {
+		t.Errorf("dave's balance is wrong: %v", err)
+	}
+}
+
+// fundingFee returns the fee estimate used for a tx with the given number of
+// inputs and the optional change output. This matches the estimate done by the
+// wallet.
+func fundingFee(numInput int, change bool) btcutil.Amount {
+	var weightEstimate input.TxWeightEstimator
+
+	// The standard fee rate used for a funding transaction.
+	var feeRate = chainfee.SatPerKWeight(12500)
+
+	// All inputs.
+	for i := 0; i < numInput; i++ {
+		weightEstimate.AddP2WKHInput()
+	}
+
+	// The multisig funding output.
+	weightEstimate.AddP2WSHOutput()
+
+	// Optionally count a change output.
+	if change {
+		weightEstimate.AddP2WSHOutput()
+	}
+
+	totalWeight := int64(weightEstimate.Weight())
+	return feeRate.FeeForWeight(totalWeight)
+}
+
+// checkChannelBalance asserts the proper response to a channel balance RPC.
+func checkChannelBalance(ht *harnessTest, node lnrpc.LightningClient,
```

Comment:
Don't we have something like that already in the `assertions.go` file?

Reply:
Yep, I am calling the functions in `assertions.go` now instead.